pytorch_lightning/utilities/auto_restart.py

```diff
@@ -441,7 +449,9 @@ def _find_current_worker(iterator: Iterator) -> Dict[str, Optional[int]]:
     return {"num_workers": num_workers, "previous_worker": previous_worker}
 
 
-def _capture_metadata_collate(samples: List, dataset: Dataset, default_collate: Callable) -> Dict:
+def _capture_metadata_collate(
+    samples: List, dataset: Dataset, default_collate: Callable, fault_tolerant_mode: _FaultTolerantMode
```

Comment:
```suggestion
    samples: List, dataset: Dataset, collate: Callable, fault_tolerant_mode: _FaultTolerantMode
```

this is not necessarily the default collate

Reply:
Done !